docs/source/array-examples.rst

```diff
@@ -0,0 +1,9 @@
+Examples
+========
+
+.. toctree::
+   :maxdepth: 1
+
+   examples/array-random.rst
+   examples/array-numpy.rst
+   examples/array-pickle.rst
```

Comment:
Sounds good. I may have bit off more than I meant to chew in this PR by reorganizing many different things, but hopefully this centralizes things for new folks coming to Dask, even if they are short-term placeholder examples. I'm happy to keep iterating on and improving the examples.